src/model.cpp

```diff
@@ -656,19 +749,56 @@ ComponentNameMap createComponentNamesMap(const ComponentPtr &component)
     return nameMap;
 }
 
-std::vector<UnitsPtr> unitsUsed(const ComponentPtr &component)
+std::vector<UnitsPtr> referencedUnits(const ModelPtr &model, const UnitsPtr &units)
+{
+    std::vector<UnitsPtr> requiredUnits;
+
+    std::string ref;
+    std::string pre;
+    std::string id;
+    double expMult;
+    double uExp;
+
+    for (size_t index = 0; index < units->unitCount(); ++index) {
+        units->unitAttributes(index, ref, pre, uExp, expMult, id);
+        if (!isStandardUnitName(ref)) {
+            auto u = model->units(ref);
+            if (u != nullptr) {
+                auto requiredUnitsUnits = referencedUnits(model, u);
+                requiredUnits.insert(requiredUnits.end(), requiredUnitsUnits.begin(), requiredUnitsUnits.end());
+                requiredUnits.push_back(u);
+            }
+        }
+    }
```

Comment:
We really ought to use more meaningful variable names (e.g. `u` should be replaced with `units`). This is API code, not test code.

Reply:
I don't get it ... I've just changed what I thought was a good variable name into "e" at your suggestion? Which is actually better?

Reply:
This API code, so we ought to use meaningful names while for test code we tend to use short variable names (see below).

Reply:
Okeydokes, done.